examples/kalman_filter.py

```diff
@@ -0,0 +1,416 @@
+r"""Illustrate how a Kalman-like filter can be defined on Lie groups.
+
+A generic Kalman filter class is defined for systems on Lie groups for which
+the exponential is subjective. Its use is illustrated on two localization
+problems, a linear and a non-linear one. In both cases, the propagation model
+is known, and sparse position measurements are obtained. It thus relies on a
+`model` of the system, providing the system's equations and jacobians.
+
+The former is a 1D-localization problem where the state is a 2D vector (x, v)
+made of the system's position and speed. The process writes
+:math:`(x_{i+1}, v_{i+1}) = (x_i + dt * v_i, v_i + dt * a_i)`,
+where dt is the time-step between i and i+1, and a_i a noisy acceleration
+measured by a given sensor.
+
+The latter is a 2D pose (position + orientation) estimation problem, where the
+state (R, x) is made a planar rotation and 2D position, i.e. a member of SE(2).
+The non-linear propagation writes
+:math:`(R_{i+1}, x_{i+1}) = (R_i \Omega_i, x_i + dt * R_i u_i)`,
+where :math:`\Omega_i, u_i` is the measured odometry of the system.
+The implementation follows that of the Invariant Extended Kalman Filter (IEKF)
+which was designed for such cases [BB2017].
+
+References
+----------
+.. [BB2017] Barrau, Bonnabel, "The Invariant Extended Kalman Filter as a Stable
+Observer", IEEE Transactions on Automatic Control, 2017
+https://arxiv.org/abs/1410.1465
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+from geomstats import algebra_utils
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.matrices import Matrices
+from geomstats.geometry.special_euclidean import SpecialEuclidean
+
+
+class LocalizationLinear:
+    """Class for modeling a linear 1D localization problem.
+
+    The state is made of a scalar position and scalar speed, thus a 2D vector.
+    A sensor provides acceleration inputs, while another one provides sparse
+    measurements of the position.
+    """
+
+    group = Euclidean(2)
+    dim = group.dim
+    dim_noise = 1
+    dim_obs = 1
+
+    @staticmethod
+    def propagate(state, sensor_input):
+        """Propagate with piece-wise constant acceleration and velocity."""
```

Comment:
Docstrings need parameters' descriptions
+ math equations would be helpful, using :math: